src/app/(auth)/subscriber/page.tsx

```diff
@@ -1,54 +1,75 @@
 'use client';
 
 import Button from '@/components/Button';
-import { getSubscribers } from '@/apis/subscriber';
-import Input from '@/components/Input';
+import { getSubscribers, deleteSubscriber } from '@/apis/subscriber';
 import Table from '@/components/Table';
 import { useEffect, useState } from 'react';
-import { Subscribers, TableData } from '@/utils/types';
+import { SubscriberResource, TableData } from '@/utils/types';
 import { UserGroupIcon } from '@heroicons/react/24/outline';
 import { PencilSquareIcon, TrashIcon } from '@heroicons/react/24/solid';
+import Link from 'next/link';
+import { toast } from 'react-toastify';
 
 const Page = () => {
-  const [subscribers, setSubscribers] = useState<Subscribers | undefined>(
-    undefined
-  );
+  const [subscribers, setSubscribers] = useState<
+    SubscriberResource | undefined
+  >(undefined);
 
-  useEffect(() => {
-    const fetchSubscribers = async () => {
-      setSubscribers(await getSubscribers());
-    };
-    fetchSubscribers();
-  }, []);
+  const fetchSubscribers = async () => {
+    setSubscribers(await getSubscribers());
+  };
 
   const [tableData, setTableData] = useState<TableData>({
     headers: ['Details', 'List'],
     rows: [],
   });
 
   useEffect(() => {
+    fetchSubscribers();
+    const showDeleteConfirmationModal = async (id: number) => {
+      const result = confirm(
+        'Are you sure you want to delete this subscriber?'
+      );
+      if (result) {
+        try {
+          await deleteSubscriber(id);
+          toast.success('Subscriber deleted successfully');
+          fetchSubscribers();
+        } catch (error: any) {
+          toast.error(error.response.data.message);
+        }
+      }
+    };
     if (subscribers) {
-      const rows = subscribers.data.map((subscriber: Subscribers) => {
+      const rows = subscribers.data.map((subscriber) => {
         return [
           <>
             <div>{subscriber.email}</div>
             <div>{subscriber.name}</div>
-            {/* <div className="text-gray-500">{campaign.tag.name}</div> */}
           </>,
           <>
             <div>{subscriber.designation}</div>
             <div className="text-gray-500">{subscriber.address}</div>
           </>,
           <>
-            {console.log(subscriber, '-->subscriber')}
             <div className="text-gray-500">{/* {subscriber.comments} */}</div>
           </>,
           <>
             <div className="flex justify-end">
-              <a href="#" className="text-gray-400 hover:text-indigo-700">
+              <Link
+                href={{
+                  pathname: '/subscriber/edit/',
+                  query: { id: subscriber.id },
+                }}
+                className="text-gray-400 hover:text-indigo-700"
+              >
                 <PencilSquareIcon className="h-5 w-5" />
-              </a>
-              <a href="#" className="ml-2 text-gray-400 hover:text-red-600">
+              </Link>
+              <a
```

Comment:
this is an invalid use of anchor tag. Anchor tags should be links with href attribute. If you want to have a clickable component, either use button or a simple div.

Reply:
I used the `<span>` tag.